firedrake/slate/static_condensation/scpc.py

```diff
@@ -77,14 +80,28 @@ def initialize(self, pc):
         self.residual = Function(W)
         self.solution = Function(W)
 
+        shapes = (Vc.finat_element.space_dimension(),
+                  np.prod(Vc.shape))
+        domain = "{[i,j]: 0 <= i < %d and 0 <= j < %d}" % shapes
+        instructions = """
+        for i, j
+            w[i,j] = w[i,j] + 1
+        end
+        """
+        self.weight = Function(Vc)
+        par_loop((domain, instructions), dx, {"w": (self.weight, INC)},
+                 is_loopy_kernel=True)
+        with self.weight.dat.vec as wc:
```

Comment:
I think you should move line 94 and 95 into the `forward_eliminiation` function so that we record the runtime cost of this in the right function (not in the initialisation). Or is there a reason why this has to be here?

Reply:
I did this for efficiency, since multiplying times the reciprocal can be done faster than division. If we use a linear solver with static condensation, we might expect to reuse this result, and there is no need to update it every time we call `forward_elimination`.

HybridizationPC also computes the weight in the initialization, but does not take the reciprocal, instead it uses a kernel with division.

Reply:
Ok!